src/Stratis.Bitcoin.Features.Consensus/Behaviors/ProvenHeadersConsensusManagerBehavior.cs

```diff
@@ -116,7 +116,7 @@ protected override Payload ConstructHeadersPayload(GetHeadersPayload getHeadersP
                 {
                     if (headersPayload.Headers[i] is ProvenBlockHeader phHeader)
                     {
-                        BlockHeader newHeader = this.chain.Network.Consensus.ConsensusFactory.CreateBlockHeader();
+                        var newHeader = new BlockHeader();
```

Comment:
He's converting PH to normal header.
The reason is something I told you days ago about the serialization problem we have in our design: if you have a PH (that inherits from BlockHeader) even if you cast it to a BlockHeader it will still be serialized as a ProvenBlockHeader.

Btw I'd like more having a `ToBlockHeader` method in ProvenBlockHeader class

Reply:
Yeah true @MithrilMan you did mention this some time ago, I wonder why we did not find it til now.

Reply:
because we solved this problem DB side, when we actually store play blockheader, converting to PH only before validating, unlucky this was left uncovered.

Now that we now we should add a test that ensure that we send always a normal header when creating the payload with mixed headers.